test/parallel/test-fs-cp.mjs

```diff
@@ -218,7 +218,7 @@ function nextdir() {
   assert.throws(
     () => cpSync(src, dest, mustNotMutateObjectDeep({ recursive: true })),
     {
-      code: 'ERR_FS_CP_EINVAL'
+      code: 'ERR_FS_CP_EEXIST'
```

Comment:
```suggestion
      code: 'ERR_FS_CP_EINVAL'
```
AFAICT if symlink in src points to location in dest, `cp` will throw `EINVAL` due to presumably attempting of copying directory to its subdirectory. This might be a bug in the subdirectory checking implementation ([`isSrcSubdir(resolvedSrc, resolvedDest)`](https://github.com/nodejs/node/blob/4f7d9d4f5d47113a9ec78b82766f17d11f6db329/lib/internal/fs/cp/cp-sync.js#L337)).

Directory structure before copying:
```r
├── copy_15
│   └── link -> node/test/.tmp.0/copy_16
├── copy_16
│   └── link -> node/test/.tmp.0/copy_16
```
Directory structure after performing regular `cp -r node/test/.tmp.0/copy_15 node/test/.tmp.0/copy_16`:
```r
├── copy_15
│   └── link -> node/test/.tmp.0/copy_16
├── copy_16
│   ├── copy_15
│   │   └── link -> node/test/.tmp.0/copy_16
│   └── link -> node/test/.tmp.0/copy_16
```
It's not really related to this particular PR, but i guess creating cyclic symlink like this should be allowed.